rollup.config.js

```diff
@@ -90,19 +105,57 @@ const customBabelConfig = {
   exclude: /node_modules/,
 };
 
+const basicConfig = {
+  input: './src/functions.js',
+  output: {
+    file: './build/functions.js',
+    name: 'functions',
+    format: 'umd',
+    globals: globalsConfig,
+  },
+  plugins: [babel(customBabelConfig)],
+  external: externalsConfig,
+};
+
+const { resolveIndexForComponent, ReadEachComponent } = functions;
+
+function generateComponentConfigs() {
+  const allComponentConfigs = [];
+  ReadEachComponent(folderName => {
+    const indexFilePath = resolveIndexForComponent(folderName);
+    allComponentConfigs.push({
+      ...basicConfig,
+      input: indexFilePath,
+      output: {
+        ...basicConfig.output,
+        file: `./${folderName}.js`,
+        name: folderName,
+      },
+      plugins: [...basicConfig.plugins, sass(componentSassOptions(folderName))],
+    });
+  });
+  return allComponentConfigs;
+}
```

Comment:
super cool stuff!

Could you maybe add a lot more description into your commit about how/why this works- I think that will be really valuable when looking back at this stuff.

I think it fits nicely in the commit since over-doing the commenting in the code might just be adding noise but adding this information in the commit is a great place to expect the explanation to be 👍